tests/test_autodiscover.py

```diff
@@ -116,3 +123,38 @@ def test_base_dir(self):
             Path(__file__).parent.resolve() / "test_structures" / "test_structure_1" / "components",
         ]
         self.assertEqual(sorted(dirs), sorted(expected))
+
+
+class TestFilepathToPythonModule(SimpleTestCase):
+    def test_prepares_path(self):
+        self.assertEqual(
+            _filepath_to_python_module(Path("tests.py")),
+            "tests",
+        )
+        self.assertEqual(
+            _filepath_to_python_module(Path("tests/components/relative_file/relative_file.py")),
+            "tests.components.relative_file.relative_file",
+        )
```

Comment:
What happens with trickier file names? Can they be imported even though they are not valid modules?

* relative-file.py
* relative.file.py
* _file.py
* .python-version

Not sure how exhaustive we should be here, but just worth thinking about?

Reply:
Very good question, IMO the paths should be valid python module names so as per [PEP-8](https://peps.python.org/pep-0008/#package-and-module-names):

> Modules should have short, all-lowercase names. Underscores can be used in the module name if it improves readability. Python packages should also have short, all-lowercase names, although the use of underscores is discouraged.

A few relevant discussions I've found:
- https://stackoverflow.com/questions/711884/python-naming-conventions-for-modules
- https://stackoverflow.com/questions/62878577/can-module-names-in-python-end-with-an-underscore
- https://stackoverflow.com/questions/8350853/how-to-import-module-when-module-name-has-a-dash-or-hyphen-in-it

So I'd go with the direction of documenting this restriction, rather than handling all possible variants.

Reply:
I've added a section on autodiscovery to README, along with a note on valid module names. See https://github.com/EmilStenstrom/django-components/pull/435/commits/e4950b79f78513877d4655bf410195f3e7e9ab73